File: backend/src/ips_app/ports/driven/logging/generic.py
from abc import ABC, abstractmethod

class GenericLoggingPort(ABC):
    @abstractmethod
    def error(self, tag: str, message: str, meta: dict) -> None: ...

    @abstractmethod
    def warn(self, tag: str, message: str, meta: dict) -> None: ...

    @abstractmethod
    def info(self, tag: str, message: str, meta: dict) -> None: ...

    @abstractmethod
    def debug(self, tag: str, message: str, meta: dict) -> None: ...

File: backend/internal/ports/driven/logging/generic.go
package portsdrivenlogging

import (
	"context"

	domainmodels "github.com/uwb-ips-dteti-ugm/ips-app/backend/internal/domain/models"
)

type Generic interface {
	Error(ctx context.Context, tag string, message string, meta domainmodels.LogMeta)
	Warn(ctx context.Context, tag string, message string, meta domainmodels.LogMeta)
	Info(ctx context.Context, tag string, message string, meta domainmodels.LogMeta)
	Debug(ctx context.Context, tag string, message string, meta domainmodels.LogMeta)
}
